engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManager.scala

```diff
@@ -168,7 +112,79 @@ class K8sDeploymentManager(modelData: ModelData, config: K8sDeploymentManagerCon
 
   override def findJobStatus(name: ProcessName): Future[Option[ProcessState]] = {
     val mapper = new K8sDeploymentStatusMapper(processStateDefinitionManager)
-    k8s.listSelected[ListResource[Deployment]](labelSelectorForName(name)).map(_.items).map(mapper.findStatusForDeployments)
+    k8s.listSelected[ListResource[Deployment]](requirementForName(name)).map(_.items).map(mapper.findStatusForDeployments)
+  }
+
+  //TODO: use https://kubernetes.io/docs/reference/using-api/server-side-apply/ in the future
+  private def createOrUpdate[O<:ObjectResource](data: O)
+                                               (implicit fmt: Format[O], rd: ResourceDefinition[O], lc: LoggingContext): Future[O] = {
+    k8s.getOption[O](data.name).flatMap {
+      case Some(_) => k8s.update(data)
+      case None => k8s.create(data)
+    }
+  }
+
+  protected def configMapForData(processVersion: ProcessVersion, deploymentData: ProcessDeploymentData): ConfigMap = {
```

Comment:
I wonder if we could extract some method like `K8sUtils.appendHash(configMap, inputForHash)` to not obfuscate logic of our deployment with k8s stuff. WDYT?

Reply:
I refactored a bit but didn't create such a method, as id generation is a bit specific anyway. Look if it's ok for you now

Reply:
It's ok